addons/hr_holidays/static/src/js/time_off_calendar/time_off_calendar_controller.js

```diff
@@ -131,9 +131,19 @@ export const TimeOffCalendarController = CalendarController.extend({
      * @private
      */
     async _onNewTimeOff() {
+        let dayContext = null;
+        if (this.mode === "day") {
+            const dateFrom = this.model.data.start_date;
+            const dateTo = dateFrom.add(1, "d");
+            dayContext = {
+                'default_date_from': dateFrom.locale('en').format('YYYY-MM-DD'),
+                'default_date_to': dateTo.locale('en').format('YYYY-MM-DD'),
+                'lang': this.context.lang,
+            }
+        }
```

Comment:
I think it would be coherent to move this whole logic in the method `_getTimeOffContext()`, what do you think? :-)

At first glance you'd even only need to override date_from with `start_date` and then keep the rest as it is